source/mesh_refinement/compaction_length.cc

```diff
@@ -40,7 +40,12 @@ namespace aspect
       if (this->get_dof_handler().n_locally_owned_dofs() == 0)
         return;
 
-      const Quadrature<dim> quadrature(this->get_fe().base_element(this->introspection().base_elements.compositional_fields).get_unit_support_points());
+      // Use a quadrature in the support points of the porosity to compute the
+      // compaction length at:
+      const typename Simulator<dim>::AdvectionField porosity = Simulator<dim>::AdvectionField::composition(
+                                                                 this->introspection().compositional_index_for_name("porosity"));
+      const Quadrature<dim> quadrature(this->get_fe().base_element(porosity.base_element(this->introspection())).get_unit_support_points());
```

Comment:
I find this hard to read, especially because of the duplicate naming of `base_element`. Can you maybe take this apart into:
```
const unsigned int base_element_index = porosity.base_element(this->introspection());
...